app/src/org/commcare/views/widgets/CommCareAudioWidget.java

```diff
@@ -105,88 +115,180 @@ public IAnswerData getAnswer() {
+            playbackTimer.cancel();
+            playbackTimer = null;
+        }
+    }
+
+    private void updatePlaybackInfo() {
+        if (player.isPlaying()) {
+            int mCurrentPosition = player.getCurrentPosition();
+            playbackSeekBar.setProgress(mCurrentPosition / 1000);
+            playbackTime.setText(getTimeString(mCurrentPosition));
+        }
+    }
+
+
+    private String getTimeString(long millis) {
+        StringBuffer buf = new StringBuffer();
+
+        int hours = (int)(millis / (1000 * 60 * 60));
+        int minutes = (int)((millis % (1000 * 60 * 60)) / (1000 * 60));
+        int seconds = (int)(((millis % (1000 * 60 * 60)) % (1000 * 60)) / 1000);
+
+        if (hours > 0) {
+            buf.append(String.format(Locale.getDefault(), "%02d", hours)).append(":");
+        }
+
+        buf.append(String.format(Locale.getDefault(), "%02d", minutes)).append(":")
+                .append(String.format(Locale.getDefault(), "%02d", seconds));
+
+        return buf.toString();
     }
 
     private void pauseAudioPlayer() {
         player.pause();
         mPlayButton.setBackgroundResource(R.drawable.play);
         mPlayButton.setOnClickListener(v -> resumeAudioPlayer());
+        stopPlaybackTimer();
     }
 
     private void resumeAudioPlayer() {
         player.start();
         mPlayButton.setBackgroundResource(R.drawable.pause);
         mPlayButton.setOnClickListener(v -> pauseAudioPlayer());
+        startPlaybackTimer();
     }
 
     private void resetAudioPlayer() {
-        player.release();
-        mPlayButton.setBackgroundResource(R.drawable.play);
-        mPlayButton.setOnClickListener(v -> playAudio());
+        if (player != null) {
+            player.release();
+        }
+        stopPlaybackTimer();
     }
 
     @Override
     protected void togglePlayButton(boolean enabled) {
         if (enabled) {
-            mPlayButton.setBackgroundResource(R.drawable.play);
+            initAudioPlayer();
         } else {
-            mPlayButton.setBackgroundResource(R.drawable.play_disabled);
+            resetAudioPlayer();
+            hidePlaybackIndicators();
         }
-        mPlayButton.setEnabled(enabled);
+    }
+
+    private void hidePlaybackIndicators() {
+        mPlayButton.setVisibility(INVISIBLE);
+        playbackSeekBar.setVisibility(INVISIBLE);
+        playbackDuration.setVisibility(INVISIBLE);
+        playbackTime.setVisibility(INVISIBLE);
+    }
+
+    private void initAudioPlayer() {
+        mPlayButton.setVisibility(VISIBLE);
+        mPlayButton.setBackgroundResource(R.drawable.play);
+        mPlayButton.setOnClickListener(v -> playAudio());
+
+        Uri filePath = Uri.parse(mInstanceFolder + mBinaryName);
+        player = MediaPlayer.create(getContext(), filePath);
+        player.setOnCompletionListener(mp -> onCompletePlayback());
+
+        playbackDuration.setVisibility(VISIBLE);
+        playbackDuration.setText("/" + getTimeString(player.getDuration()));
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dimagi/commcare-android/blob/0853883e2914ad929d254cf69030a86e1bd7a364/app/src/org/commcare/views/widgets/CommCareAudioWidget.java#L252"></span>Do not concatenate text displayed with <code>setText</code>. Use resource string with placeholders.</td>
    </tr>
  </tbody>
</table>